tests/test_utils.py

```diff
@@ -19,3 +30,89 @@ def test_route_params():
     pass_filter = set()
     out = route_params(params, destination, pass_filter)
     assert out["foo"] is params["model__foo"]
+
+
+def test_compile_with_params_nesting():
```

Comment:
What does this test that `tests/test_compile_kwargs.py` doesn't?

Reply:
We should be able to remove them. The coverage is the same. This is mainly checking that the returned container types are not altered. We can remove this and `test_compile_with_params_dependency` if you don't think we should/will make  `compile_with_params` public (if we're going to make it public, we should keep these).

Reply:
Why would users use `compile_with_params`? I'm not seeing a reason.

Reply:
I mostly agree. I can't imagine any _good_ use cases. Should we remove these tests then?

Reply:
I went ahead and removed them. I think it's the right thing to do here.